### PR TITLE
4 get review by

### PR DESCRIPTION
**Pull request by tommarsh27**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -21,12 +20,52 @@ describe('/api/categories', () => {
             .expect(200)
             .then(({body}) => {
                 expect(Array.isArray(body.categories)).toBe(true)
-                expect(body.categories.length > 0)
+                expect(body.categories.length = 4)
```
Comment (HMelesi) [inline __tests__/app.test.js]: Not sure this is doing what you want it to be doing? Refactor to use .toBe(4)

File: __tests__/app.test.js [hunk 2]
```diff
@@ -21,12 +20,52 @@ describe('/api/categories', () => {
...
+})
+
+describe('/api/reviews', () => {
+    describe('GET by review_id', () => {
+        test('200: responds with an object containing a key of reviews and a value of an array containing the review with the provided review_id', () =>{
+            return request(app)
+            .get('/api/reviews/1')
+            .expect(200)
+            .then(({body}) => {
+                expect(body.reviews).toEqual([{
```
Comment (HMelesi) [inline __tests__/app.test.js]: We would want this review back as an object on a key of 'review', so the body would look something like:
`{
  review: {
    review_id: 1,
    ...
  }
}`

Comment (HMelesi) [reply __tests__/app.test.js]: Right now you have it on a key of 'reviews' and its an array of length 1 - just a bit of data manipulation in your api needed.

File: app.js [hunk 3]
```diff
@@ -1,13 +1,14 @@
 const express = require('express')
-const { getCategories } = require('./controllers/controller')
+const { getCategories } = require('./controllers/categories_controller')
+const { getReveiws } = require('./controllers/reviews_controller')
```
Comment (HMelesi) [inline app.js]: Typo on getReviews here

File: app.js [hunk 4]
```diff
@@ -16,6 +17,12 @@ app.use((err, req, res, next) => {
     }
 })
 
+app.use((err, req, res, next) => {
+    if(err.code === '22P02') {
+        res.status(400).send({msg: 'Bad Request'})
+    }
```
Comment (HMelesi) [inline app.js:25]: You'll need to add a:
` else {
  next(err)
}`
here if you want to see any other errors in your console.log on line 27!

**Review by HMelesi — CHANGES_REQUESTED** [review]
Have a look at the changes requested, any questions let me know!

File: app.js (hunk 4 above)
Comment (tommarsh27) [reply app.js:25]: done

File: app.js (hunk 3 above)
Comment (tommarsh27) [reply app.js]: Done, amusingly it still worked because I made the same typo in the controller lol

File: __tests__/app.test.js (hunk 1 above)
Comment (tommarsh27) [reply __tests__/app.test.js]: done

File: __tests__/app.test.js (hunk 2 above)
Comment (tommarsh27) [reply __tests__/app.test.js]: done

**Review by HMelesi — APPROVED** [review]
Looks good - one minor thing, you've called it getReviews but it's just getting one review, I'd probably change that name. But no need to submit a new PR.

**Pull request merged**